### ISSUE TITLE
way to disable 404 route

### ISSUE THREAD

**Issue by harrysolovay** [denoland/fresh#1918]
Follow-up from #1916

Might it be valuable to provide a way to disable the 404 route?

**Comment by lionel-rowe** [comment]
FWIW, you can work around it fairly easily by checking the `status` of the `Response` object returned from `freshHandler`:

```ts
const freshHandler = await createHandler(manifest, config);

Deno.serve(async (req, info) => {
  const { pathname } = new URL(req.url);

  if (pathname === "/special-route") {
    return new Response('special route');
  }

  const res = await freshHandler(req, info);

  if (res.status === 404) {
    return new Response('overridden 404 response', { status: 404 });
  }

  return res;
});
```

Only drawback is that the Fresh 404 page is still rendered (i.e. slight performance hit, and any side effects from rendering it will still happen).

**Comment by harrysolovay** [comment]
Good idea! If the 404 page is empty, this shouldn't be too bad. Still, a way to opt out would be imo valuable.

**Comment by deer** [comment]
@harrysolovay @lionel-rowe, what's the expected behavior here? I've modified `RouterOptions` so it looks like this:
```ts
export interface RouterOptions {
  trailingSlash?: boolean;
  ignoreFilePattern?: RegExp;
  disable404?: boolean;
}
```
So you can now use this (at least in my branch on my computer) to disable the 404 page. But what does that mean exactly? Right now I have this diff somewhere within the server code:
```diff
               async renderNotFound<Data = undefined>(data: Data) {
+                if (disable404) {
+                  throw new Deno.errors.NotFound();
+                }
                 return await renderNotFound(req, params, ctx, data);
               },
```
So if the flag is true, instead of running the `renderNotFound` handler, it will throw a not found error. Is this what you're looking for? I guess that error can then be caught and dealt with on your own.

**Comment by harrysolovay** [comment]
> Is this what you're looking for?

Yes.

> I guess that error can then be caught and dealt with on your own.

This would be a fine solution, although returning undefined may be preferable imo.

**Comment by marvinhagemeister** [comment]
Reopening, because I found a better way to do it in https://github.com/denoland/fresh/pull/3086
